prov/acmp/src/acmp.c

```diff
@@ -1774,6 +1845,7 @@ acmp_resolve_dest(struct acmp_ep *ep, struct acm_msg *msg, uint64_t id)
 		acm_log(2, "have address, resolving route\n");
 		acm_increment_counter(ACM_CNTR_ADDR_CACHE);
 		atomic_inc(&ep->counters[ACM_CNTR_ADDR_CACHE]);
+		acmp_acquire_sgid(saddr, dest);
```

Comment:
I'm not objecting to converting a source IP/IPv6 address into an SGID.  However, the SGID that gets stored with the acmp_dest should not depend on which request gets processed by ibacm first.  That sort of assignment should be independent of any request.  I.e. how do you know that a second resolve operation uses the same IP address?  Or that the IP -> SGID mapping is the same?  How does this change handle something like: src=10.0.0.1 dst = 10.0.1.2 followed by src=10.0.0.2 dst=10.0.1.2?

It seems that either the acmp_dest mapping to an SGID is unique, in which case we can determine it earlier, such as when the acmp_dest is allocated, and we don't need the saddr as part of the process.  Or the acmp_dest mapping to an SGID is not unique, in which case, it isn't helpful to store the SGID with the acmp_dest.